src/api/nodeRpcCall.js

```diff
@@ -1,139 +1,148 @@
+          passportNumber: 'Encryptable',
+          signingAbility: 'Encryptable', // FIXME enum
+          signingAbilityConditions: 'Encryptable',
+        },
+        CompanyData: {
+          name: 'Text',
+          // Truthful scope of business
+          purpose: 'Text',
+          logoURL: 'Text',
+          charterURL: 'Text',
+          totalCapitalAmount: 'Text', // FIXME integer instead? Will have issues with decimals though.
+          totalCapitalCurrency: 'Text', // FIXME maybe some enum?
+          numberOfShares: 'Text', // FIXME integer instead? Are fractional shares supported
+          valuePerShare: 'Text', // FIXME same as totalCapitalAmount probably
+          // History of transfer of shares
+          history: 'Text', // FIXME array of well defined structs?
+          brandNames: 'Vec<BrandName>',
+          onlineAddresses: 'Vec<OnlineAddress>',
+          physicalAddresses: 'ec<PhysicalAddress>',
+          statutoryOrganMembers: 'Vec<Person>',
+          principals: 'Vec<Principal>',
+          shareholders: 'Vec<Shareholder>',
+          UBOs: 'Vec<UBO>',
+        },
       },
-    }
-  });
+    });
+  }
   return __apiCache;
 };
 
-const crossReference = (api, blockchainData, allCentralizedData, motions, isReferendum) =>
-  blockchainData.map((item) => {
-    const proposalHash = isReferendum ? item.imageHash : (
-      item.boundedCall?.Lookup?.hash_ ??
-      item.boundedCall?.Legacy?.hash_
-    )
+// eslint-disable-next-line max-len
+const crossReference = (api, blockchainData, allCentralizedData, motions, isReferendum) => blockchainData.map((item) => {
+  const proposalHash = isReferendum ? item.imageHash : (
+    item.boundedCall?.Lookup?.hash_
+      ?? item.boundedCall?.Legacy?.hash_
+  );
 
-    const centralizedDatas = allCentralizedData.filter((cItem) => ( cItem.hash === proposalHash))
-    const blacklistMotionHash = api.tx.democracy.blacklist(
-      proposalHash,
-      isReferendum ? item.index : null
-    ).method.hash.toString()
+  const centralizedDatas = allCentralizedData.filter((cItem) => (cItem.hash === proposalHash));
+  const blacklistMotionHash = api.tx.democracy.blacklist(
+    proposalHash,
+    isReferendum ? item.index : null,
+  ).method.hash.toString();
 
-    return {
-      ...item,
-      centralizedDatas,
-      blacklistMotion: motions.includes(blacklistMotionHash) ? blacklistMotionHash : null
-    }
-  }
-);
+  return {
+    ...item,
+    centralizedDatas,
+    blacklistMotion: motions.includes(blacklistMotionHash) ? blacklistMotionHash : null,
+  };
+});
 
-const submitExtrinsic = async (extrinsic, walletAddress) => {
-  const api = await getApi();
-  const { signer } = await web3FromAddress(walletAddress);
-  return await new Promise((resolve, reject) => {
+const submitExtrinsic = async (extrinsic, walletAddress, api, signerFromAdress = true) => {
+  const { signer } = signerFromAdress ? await web3FromAddress(walletAddress) : await web3FromSource('polkadot-js');
```

Comment:
I dont understand what signerfromAddress false is handling.

Reply:
@SzymonZiebaNeti I think we went through it on call and decided that we should always use `await web3FromAddress(walletAddress)` for everything.

Reply:
Could you also explain it to me why this way of doing it was considered ?

Reply:
The difference is:
* `web3FromAddress(walletAddress)` takes signer for given wallet address from any extension-wallet,
* `web3FromSource('polkadot-js')` takes default wallet from polkadot.js extension.

Hardcoding polkadot-js is obviously wrong, as we want to support other wallets as well.

It wasn't really "considered". @SzymonZiebaNeti just followed how existing code worked during refactor. If you take a look at original code, some functions use `web3FromSource` (https://github.com/liberland/liberland_frontend/blob/20bd59fe38b2faecc9252d61e435069914e57377/src/api/nodeRpcCall.js#L564) and some use `web3FromAddress`.